timeseries/src/autogluon/timeseries/trainer/abstract_trainer.py

```diff
@@ -461,66 +503,80 @@ def _train_multi(
                     train_data, model=model, val_data=val_data, time_limit=time_left
                 )
 
+        logger.info("\nTraining complete")
+        logger.info(f"Models trained: {model_names_trained}")
+
+        try:
+            best_model = self.get_model_best()
+            logger.info(f"Best model: {best_model}")
+            logger.info(
+                f"Best model score: {self.get_model_attribute(best_model, 'val_score')}"
+            )
+        except ValueError as e:
+            logger.error(str(e))
+
         return model_names_trained
 
     def leaderboard(self, data: Optional[TimeSeriesDataFrame] = None) -> pd.DataFrame:
-        logger.log(30, "Generating leaderboard for all models trained...")
-        model_names = self.get_model_names()
-        score_val = []
-        fit_order = list(range(1, len(model_names) + 1))
-        score_dict = self.get_models_attribute_dict("val_score")
-
-        fit_time_marginal = []
-        fit_time_marginal_dict = self.get_models_attribute_dict("fit_time")
+        logger.debug("Generating leaderboard for all models trained")
 
-        for model_name in model_names:
-            score_val.append(score_dict[model_name])
-            fit_time_marginal.append(fit_time_marginal_dict[model_name])
+        model_names = self.get_model_names()
+        model_info = {}
+        for ix, model_name in enumerate(model_names):
+            model_info[model_name] = {
+                "model": model_name,
+                "fit_order": ix + 1,
+                "score_val": self.get_model_attribute(model_name, "val_score"),
+                "fit_time_marginal": self.get_model_attribute(model_name, "fit_time"),
+                "predict_time": self.get_model_attribute(model_name, "predict_time"),
+            }
 
-        test_score = []
         if data is not None:
-            logger.log(
-                30, "Additional data provided, testing on the additional data..."
+            logger.info(
+                "Additional data provided, testing on additional data. Resulting leaderboard "
+                "will be sorted according to test score (`score_test`)."
             )
             for model_name in model_names:
-                test_score.append(self.score(data, model_name))
-        df = pd.DataFrame(
-            data={
-                "model": model_names,
-                "score_val": score_val,
-                "fit_time_marginal": fit_time_marginal,
-                "fit_order": fit_order,
-            }
-        )
-        if test_score:
-            df["score_test"] = test_score
+                try:
+                    model_info[model_name]["score_test"] = self.score(data, model_name)
+                except Exception as e:  # noqa
+                    logger.error(
+                        f"Cannot score with model {model_name}. An error occurred: {str(e)}"
+                    )
+                    model_info[model_name]["score_test"] = float("nan")
```

Comment:
Please add pred_time_test by splitting up score into predict and evaluate, and timing predict.

Note that this logic will stop working in an ideal fashion once we add ensembling. (It will still work, but will redundantly predict with models that are part of ensembles.) To do it optimally we will need to pull in the Tabular logic of DAG-based model prediction. (doesn't have to be resolved in this PR, can be done post-release)

Reply:
I will skip this for now as it also depends on the evaluation changes coming soon.